### ISSUE TITLE
OpenSSL 3-beta issue on mac

### ISSUE THREAD

**Issue by jrgleason** [sfackler/rust-openssl#1487]
I am having issues getting the openssl integration working. I am on OSX and my OpenSSL is installed from source. After completed I `export OPENSSL_DIR=/usr/local/ssl` but when I run `./x.py install cargo` I get

>thread 'main' panicked at 'OpenSSL library directory does not exist: /usr/local/ssl/lib'

What directory should I be using?

**Comment by jrgleason** [comment]
More info https://stackoverflow.com/questions/68019661/how-do-i-install-the-development-headers-when-compiling-openssl-from-source-on-m

**Comment by sfackler** [comment]
Does `/usr/local/ssl/lib` exist?

**Comment by jrgleason** [comment]
@sfackler No I don't know if it gets installed somewhere else by defailt...

```
% ls -al /usr/local/ssl 
total 64
drwxr-xr-x   9 root  wheel    288 Jun 17 08:47 .
drwxr-xr-x  10 root  wheel    320 Jun 16 22:13 ..
drwxr-xr-x   2 root  wheel     64 Jun 16 22:13 certs
-rw-r--r--   1 root  wheel    412 Jun 16 22:13 ct_log_list.cnf
-rw-r--r--   1 root  wheel    412 Jun 17 08:47 ct_log_list.cnf.dist
drwxr-xr-x   5 root  wheel    160 Jun 17 08:47 misc
-rw-r--r--   1 root  wheel  11808 Jun 16 22:13 openssl.cnf
-rw-r--r--   1 root  wheel  11808 Jun 17 08:47 openssl.cnf.dist
drwxr-xr-x   2 root  wheel     64 Jun 16 22:13 private
```

The headers are in `/usr/local/include/openssl` but I can't find where a lib directory would be getting installed. I am used to Linux so it might just be a diff folder.

**Comment by sfackler** [comment]
It seems like you set the install prefix to `/usr/local`, so you'd want to set OPENSSL_DIR to that. The crate is expecting to find a directory layout that looks like `$OPENSSL_DIR/lib/libssl.so`, `$OPENSSL_DIR/include/openssl/opensslv.h`, etc.

**Comment by jrgleason** [comment]
@sfackler Out I tried that before but let me try again. In the mean time here is the make install output from openssl...

[install.txt](https://github.com/sfackler/rust-openssl/files/6670304/install.txt)

**Comment by jrgleason** [comment]
Getting 

```
You can try fixing this setting the `OPENSSL_DIR` environment variable
  pointing to your OpenSSL installation or installing OpenSSL headers package
  specific to your distribution:
```
  
Even though 

```
rust % echo $OPENSSL_DIR
/usr/local
```
Is there something else I have to do to have it see the edited env var?

**Comment by jrgleason** [comment]
@sfackler FYI that all looks correct...

```
ls /usr/local/include/openssl 
aes.h			dsa.h			pemerr.h
asn1.h			dsaerr.h		pkcs12.h
```

```
rust % ls /usr/local/lib/
bash			libcrypto.dylib		libuv.dylib
docker			libssl.3.dylib		libuv_a.a
dtrace			libssl.a		node
engines-3		libssl.dylib		node_modules
libcrypto.3.dylib	libuv.1.0.0.dylib	ossl-modules
libcrypto.a		libuv.1.dylib		pkgconfig
```

**Comment by sfackler** [comment]
That install output and directory layout looks right to me. What is the full output from the build script that's failing? It should say what is is seeing OPENSSL_DIR set to.

**Comment by jrgleason** [comment]
I will output shortly I tried reverting any changes from git and reset the config now I am going to try again I will let you know.

**Comment by jrgleason** [comment]
also just FYI the version of openssl is a beta version 

```
openssl version
OpenSSL 3.0.0-beta1-dev  (Library: OpenSSL 3.0.0-beta1-dev )
```

**Comment by sfackler** [comment]
I think you're still running into an earlier error, but the current release of this crate does not support OpenSSL 3.0.0. Support is on a branch, which we'll land when the 3.0.0 beta releases. In the meantime, you'll probably want to use a build of 1.1.1k.

**Comment by jrgleason** [comment]
@sfackler is 1.1.1k a rust version or OpenSSL one? Would I be able to downgrade to OpenSSL 2.x?

**Comment by sfackler** [comment]
An OpenSSL version. There is no OpenSSL 2.x release - 1.1.1 is the current stable and they're skipping entirely over 2.x into 3.0.0 for some reason. https://www.openssl.org/source/

**Comment by jrgleason** [comment]
I will keep this open till I can confirm but sounds legit to me.

**Comment by jrgleason** [comment]
@sfackler do you have a link for the rust issue so I can track when I can try 3 again?

**Comment by sfackler** [comment]
https://github.com/sfackler/rust-openssl/pull/1264

**Comment by jrgleason** [comment]
Looks like I can confirm that downgrading OpenSSL to "OpenSSL 1.1.1l-dev  xx XXX xxxx" fixes the issue I will keep an eye out on the other issue and see if there is anywhere I can test/help. Thanks!
